Sources/MapboxNavigation/NavigationViewController.swift

```diff
@@ -418,6 +418,7 @@ open class NavigationViewController: UIViewController, NavigationStatusPresenter
                               didPassVisualInstructionPoint: firstInstruction,
                               routeProgress: navigationService.routeProgress)
         }
+        navigationMapView?.simulatesLocation = (navigationService.locationManager is SimulatedLocationManager) || (navigationService.locationManager is ReplayLocationManager)
```

Comment:
NavigationLocationManager is `open`, so we should avoid special-casing SimulatedLocationManager and ReplayLocationManager if possible, because the developer is free to implement their own subclass that may or may not simulate location.

Reply:
Thanks for the suggestion, I add the property of the `simulatesLocation` to `NavigationLocationManager` to control the `navigationMapView.simulatesLocation` through the `navigationService.locationManager`. And also when the simulation mode is `.onPoorGPS`, it would still be `true` to display the motion of `userLocationStyle` based on route controller.